src/language/ast.js

```diff
@@ -44,6 +44,30 @@ export type Location = {
 };
 
 /**
+ * Represents the different kinds of tokens in a GraphQL document.
+ */
+export type TokenKind = '<SOF>'
```

Comment:
I'm not sure if there's a support for that yet - we probably need to do in a verbose way for the time being.
Another thing I noticed is that `ast.js` uses `Token` object solely to typecheck `Location` object - we could potentially move `export type Token` to `lexer.js` and import the type from there.